File: Actividad 5 Paso de Arreglos a Métodos o Funciones/Actividad5.java
public class Actividad5 {

    public static void imprimirArreglo(int[] arr) {

        for (int elemento : arr) {
            System.out.print(elemento + " ");
        }

        System.out.println();
    }

    public static int encontrarMaximo(int[] arr) {

        int maximo = arr[0];

        for (int elemento : arr) {

            if (elemento > maximo) {
                maximo = elemento;
            }
        }

        return maximo;
    }

    public static int encontrarMinimo(int[] arr) {

        int minimo = arr[0];

        for (int elemento : arr) {

            if (elemento < minimo) {
                minimo = elemento;
            }
        }

        return minimo;
    }

    public static int[] invertirArreglo(int[] arr) {

        int izquierda = 0;
        int derecha = arr.length - 1;

        while (izquierda < derecha) {

            int temp = arr[izquierda];
            arr[izquierda] = arr[derecha];
            arr[derecha] = temp;

            izquierda++;
            derecha--;
        }

        return arr;
    }

    public static void main(String[] args) {

        int[] numeros = {5, 8, 2, 9, 1, 7, 4, 6, 3, 10};

        System.out.println("Arreglo original:");
        imprimirArreglo(numeros);

        System.out.println("\nValor máximo: " + encontrarMaximo(numeros));
        System.out.println("Valor mínimo: " + encontrarMinimo(numeros));

        invertirArreglo(numeros);

        System.out.println("\nArreglo invertido:");
        imprimirArreglo(numeros);
    }
}

File: Actividad 5 Paso de Arreglos a Métodos o Funciones/Actividad5.py
def imprimirArreglo(arr):

    for elemento in arr:
        print(elemento, end=" ")

    print()


def encontrarMaximo(arr):

    return max(arr)


def encontrarMinimo(arr):

    return min(arr)


def invertirArreglo(arr):

    izquierda = 0
    derecha = len(arr) - 1

    while izquierda < derecha:

        arr[izquierda], arr[derecha] = arr[derecha], arr[izquierda]

        izquierda += 1
        derecha -= 1

    return arr


# Programa principal
numeros = [5, 8, 2, 9, 1, 7, 4, 6, 3, 10]

print("Arreglo original:")
imprimirArreglo(numeros)

print("\nValor máximo:", encontrarMaximo(numeros))
print("Valor mínimo:", encontrarMinimo(numeros))

invertirArreglo(numeros)

print("\nArreglo invertido:")
imprimirArreglo(numeros)
